commit.go

```diff
@@ -371,8 +369,8 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
 		}
 	}
 
-	img, err := is.Transport.GetStoreImage(b.store, dest)
-	if err != nil && !errors.Is(err, storage.ErrImageUnknown) {
+	_, img, err := is.ResolveReference(dest)
+	if err != nil && err.Error() != fmt.Sprintf("trying to resolve a non-%s: reference %q", is.Transport.Name(), transports.ImageName(dest)) {
```

Comment:
Errors coming back from the new API can't be checked using `errors.Is()` or `errors.As()`?  That seems like a step backward.

Reply:
I would expect `errors.Is(err, istorage.ErrNoSuchImage)` to work, and it seems to have worked correctly in CRI-O.

Reply:
That’s not an “image not found” path, that’s “invalid use of this API” path, though.

I definitely need to fix the documentation and make the error type a maintained API guarantee, I’ll do that today.

Reply:
Oops, I’m being stupid everywhere in this PR review, sorry about that.

I thought this is an “image not found” check, but this really is trying to use the API with a non-c/storage reference.

It seems to me that if `Commit` is called with a non-c/storage destination, the following code touching `img.ID` and `RemoveNames` and `IIDFile` should not be entered at all, because the image was not written there and this is modifying something unrelated to the commit. So; the caller here should check for a non-c/storage destination and not call `ResolveReference` at all in those cases.

Reply:
@mtrmac I have tried following diff and it does not works as it should.

```diff
diff --git a/commit.go b/commit.go
index fba35b2c3..6d4c27c78 100644
--- a/commit.go
+++ b/commit.go
@@ -3,6 +3,7 @@ package buildah
 import (
        "context"
        "encoding/json"
+       "errors"
        "fmt"
        "io"
        "os"
@@ -370,7 +371,7 @@ func (b *Builder) Commit(ctx context.Context, dest types.ImageReference, options
        }
 
        _, img, err := is.ResolveReference(dest)
-       if err != nil && err.Error() != fmt.Sprintf("trying to resolve a non-%s: reference %q", is.Transport.Name(), transports.ImageName(dest)) {
+       if err != nil && !errors.Is(err, is.ErrNoSuchImage) {
                return imgID, nil, "", fmt.Errorf("locating image %q in local storage: %w", transports.ImageName(dest), err)
        }
        if err == nil {
diff --git a/tests/test_runner.sh b/tests/test_runner.sh
index a93654b37..021e2dd27 100755
--- a/tests/test_runner.sh
+++ b/tests/test_runner.sh
@@ -19,4 +19,4 @@ function execute() {
 TESTS=${@:-.}
 
 # Run the tests.
-execute time bats --tap $TESTS
+execute time bats --tap $TESTS --filter "\"blobcache-commit\""
```

To run the test
```console
sudo ./tests/test_runner.sh
```

Reply:
The error which fails due to exact this match

```console
# Error: committing container "pause-working-container" to "dir:/var/tmp/buildah_tests.sv7ziw/doomed": locating image "dir:/var/tmp/buildah_tests.sv7ziw/doomed" in local storage: trying to resolve a non-containers-storage: reference "dir:/var/tmp/buildah_tests.sv7ziw/doomed"
# [ rc=125 (** EXPECTED 0 **) ]
```

Reply:
@flouthoc Yeah, I was being an idiot. See above, and elsewhere about `dest2`.